provisioner.sh

```diff
@@ -1,26 +1,49 @@
 #!/bin/bash
 
-# Step 1. Install openfaas-cli
-curl -sSL https://cli.openfaas.com | sudo -E sh
-
-# Step 2. Install microk8s and configure it
-# This step will add all the necessary packages
-# And permissions to use microk8s as needed
-sudo snap install microk8s --classic
-sudo usermod -a -G microk8s vagrant
-sudo chown -f -R vagrant ~/.kube
-newgrp microk8s
-microk8s status --wait-ready
-microk8s enable dns dashboard storage registry
-microk8s start
-
-# Step 3. Clone OpenFaaS repo in order to use it
-git clone https://github.com/openfaas/faas-netes
-cd faas-netes
-
-# Step 4. Create the OpenFaaS namespaces for the functions
-microk8s kubectl apply -f ./namespaces.yml
-microk8s kubectl apply -f ./yaml
-
-# Step 5. Create a local Docker container registry 
-docker run -d -p 5000:5000  --restart=always --name registry registry:2
\ No newline at end of file
+set -e
+
+function install_microk8s () {
+  # Install microk8s
+  sudo snap install microk8s --classic
+
+  # Start the service with all modules enabled
+  microk8s status --wait-ready
+  microk8s enable dns dashboard storage registry
+  microk8s start
+}
+
+function microk8s_permissions () {
+  # Add all the necessary packages and permissions to use microk8s
+  sudo usermod -a -G microk8s vagrant
+  sudo chown -f -R vagrant ~/.kube
+  newgrp microk8s
+}
+
+function install_openfaas () {
+  # Install openfaas-cli
+  curl -sSL https://cli.openfaas.com | sudo -E sh
+}
+
+function clone_faasnetes () {
+  # Clone the faasnetes git repo in order to create the functions
+  git clone https://github.com/openfaas/faas-netes
+  cd faas-netes
+}
```

Comment:
this one seem closely related to the next one... it's probably better to merge them together or make one call the other

This has to do with [cohesion](https://thevaluable.dev/cohesion-coupling-guide-examples/)